numpy/core/numeric.py

```diff
@@ -971,22 +971,40 @@ def tensordot(a, b, axes=2):
         * ``axes = 1`` : tensor dot product :math:`a\\cdot b`
         * ``axes = 2`` : (default) tensor double contraction :math:`a:b`
 
-    When `axes` is integer_like, the sequence for evaluation will be: first
-    the -Nth axis in `a` and 0th axis in `b`, and the -1th axis in `a` and
-    Nth axis in `b` last.
+    When `axes` is integer_like, the sequence of axes for evaluation will be: from
+    the -Nth axis to the -1th axis in `a`, and from the 0th axis to (N-1)th axis in `b`.
+    For example, ``axes = 2`` is the equal to ``axes = [[-2, -1], [0, 1]]``.
+    When N-1 is smaller than 0, or when -N is larger than -1,
+    the element of 'a' and 'b' are defined as the 'axes'.
```

Comment:
Agree with @mattip this part is a little unclear and redundant before.
Therefore, I add some details to help express more clearly.